origamiUROP/polygons.py

```diff
@@ -20,37 +23,39 @@
 
 EDGE_TYPE = {0: "boundary", 1: "scaffold", 2: "staple"}
 
-
-class Edge:
+class Edge():
     """ 
     An Edge is a combination of two vertices and stores geometric information about this connection 
     """
 
     def __init__(self, vertex_1: list, vertex_2: list, edge_type: int = 0):
         self.vertices = np.array([vertex_1, vertex_2])
-        self.type = EDGE_TYPE[edge_type]
-        ## DEFAULT INITIALISED VALUES
-        self.dir = np.array([0.0, 0.0, 1.0])  # direction
-        self.linear_velocity = 0.0
-        self.angular_velocity = 0.0
 
-    def __len__(self):
+        # don't use type because that's a protected function in Python
+        self.kind = EDGE_TYPE[edge_type]
+        
+        # these belong to individual nucleotides not the whole edge
+        # self.linear_velocity = 0
+        # self.angular_velocity = 0
+
+    @property
+    def length(self):
         return np.linalg.norm(self.vertices[1] - self.vertices[0])
 
     @property
     def vector(self) -> np.ndarray:
         return self.vertices[1] - self.vertices[0]
 
     @property
-    def unit_vector(self):
+    def unit_vector(self) -> np.ndarray:
         return self.vector / (self.vector ** 2).sum() ** 0.5
 
 
 def define_edges(vertices_of_polygon, index_1, index_2, edge_type):
     return Edge(vertices_of_polygon[index_1, :], vertices_of_polygon[index_2, :],)
 
 
-class BoundaryPolygon:
+class BoundaryPolygon():
```

Comment:
When we call it would be great if you could quickly explain the difference of this with and without the brackets, not sure I get it atm

Reply:
just had a look and actually I think my way is actually the wrong one!

You use the brackets when subclassing e.g. `class Square(Polygon)`

I think there was a time where you would use no brackets to describe what's called an Abstract Class. This is basically a class that you never use but other classes inherit from.

However in Python 3 I don't think this is actually enforced and your method is actually preferred I think.

For me, I do three things to help with syntax which I do believe is useful:

- `class AbstractClass`
- `class ClassIWillUse()`
- `class ClassThatInheritsFromAnotherClass(AnotherClass)`